### PR TITLE
RTL: change when to set a heading setpoint, generally leave it up to executer

### PR DESCRIPTION
**Pull request by sfuhrer**
### Solved Problem
VTOLs yawing to the home position heading when descending in hover during RTL instead of weather-vaning.

### Solution
Continue with what was started in https://github.com/PX4/PX4-Autopilot/pull/22532: 
- remove `RTL_HDG_MD` (necessity of it hasn't been proven and it interferes with the here proposed solution)
- only set heading setpoint in `Navigator::RTL` once above landing point, or when RTL is triggered close to it
- never set a heading setpoint in the RTL if weather vane is enabled

### Changelog Entry
For release notes:
```
Bugfix: Fix weather vane in RTL
Feature: Make RTL behavior more deterministic by removing the parameter RTL_HDG_MD.
```

### Test coverage
SITL tested. 

### Context
This is how a normal RTL with a MC looks like (the behavior doesn't change with this PR):
- climb while keeping heading
- return over landing point without a specified heading --> executor (FlightTaskAuto) sets heading in direction of flight
- once over landing point set destination heading (home position heading in this case) and don't change it anymore

https://github.com/PX4/PX4-Autopilot/assets/26798987/394599d1-555a-4b40-92b6-a775a5bf7ffc

### CODE REVIEW

File: src/modules/navigator/rtl_direct.cpp [hunk 1]
```diff
@@ -319,7 +323,9 @@ void RtlDirect::set_rtl_item()
 
 	case RTLState::LAND: {
 
-			setLandMissionItem(_mission_item, _destination, rtl_heading_mode);
+			// set final yaw if WV is disabled
```
Comment (MaEtUgR) [inline src/modules/navigator/rtl_direct.cpp:326]:
```suggestion
			// set final yaw if weather vane is disabled
```

File: src/modules/navigator/rtl_direct.cpp [hunk 2]
```diff
@@ -255,8 +255,10 @@ void RtlDirect::set_rtl_item()
 				.yaw = _destination.yaw,
```
Comment (MaEtUgR) [inline src/modules/navigator/rtl_direct.cpp:255]: The passed on destination already has a yaw(/heading?) which now gets ignored at least in this case here 🤯

**Review by MaEtUgR — COMMENTED** [review]
Nice. Thanks for this generally nice improvement!
The only question I have is about this `DestinationPosition dest` which contains a yaw. As I read it this was supposed to contain the actual heading setpoint and I would have probably gone back to this. But if we have a separate `heading_sp` then I'd suggest only having one way. Or make it very clear when to use which.

**Comment by MaEtUgR** [comment]
@sfuhrer I rebased and added a commit which I'd presume does the same without the duplication in the interface. Did I overlook something?

File: src/modules/navigator/rtl_direct.cpp (hunk 2 above)
Comment (sfuhrer) [reply src/modules/navigator/rtl_direct.cpp]: correct - let's go with your proposed solution.

File: src/modules/navigator/mission_block.cpp [hunk 3]
```diff
@@ -946,8 +945,9 @@ void MissionBlock::setLoiterToAltMissionItem(mission_item_s &item, const Destina
 	item.loiter_radius = loiter_radius;
 }
 
-void MissionBlock::setLoiterHoldMissionItem(mission_item_s &item, const DestinationPosition &dest, float loiter_time,
-		float loiter_radius, HeadingMode heading_mode) const
+void MissionBlock::setLoiterHoldMissionItem(mission_item_s &item, const PositionYawSetpoint &pos_yaw_sp,
+		float loiter_time,
+		float loiter_radius) const
```
Comment (MaEtUgR) [inline src/modules/navigator/mission_block.cpp:950]:
```suggestion
		float loiter_time, float loiter_radius) const
```

**Review by MaEtUgR — APPROVED** [review]
Much clearer 👍

**Review by MaEtUgR — APPROVED** [review]

**Pull request merged**